atl-checker/src/algorithms/certain_zero/search_strategy/linear_programming_search.rs

```diff
@@ -0,0 +1,224 @@
+use crate::algorithms::certain_zero::search_strategy::linear_constrained_phi::{
+    ConstrainedPhiMapper, LinearConstrainedPhi,
+};
+use crate::algorithms::certain_zero::search_strategy::linear_constraints::{
+    ComparisonOp, LinearConstraint,
+};
+use crate::algorithms::certain_zero::search_strategy::{SearchStrategy, SearchStrategyBuilder};
+use crate::atl::Phi;
+use crate::edg::atlcgsedg::AtlVertex;
+use crate::edg::Edge;
+use crate::game_structure::lcgs::ast::DeclKind;
+use crate::game_structure::lcgs::ir::intermediate::{IntermediateLcgs, State};
+use crate::game_structure::lcgs::ir::symbol_table::SymbolIdentifier;
+use minilp::{LinearExpr, OptimizationDirection, Problem, Variable};
+use priority_queue::PriorityQueue;
+use std::collections::HashMap;
+use std::option::Option::Some;
+use std::sync::Arc;
+
+/// A SearchStrategyBuilder for building the LinearProgrammingSearch strategy.
+pub struct LinearProgrammingSearchBuilder {
+    pub game: IntermediateLcgs,
+}
+
+impl SearchStrategyBuilder<AtlVertex, LinearProgrammingSearch> for LinearProgrammingSearchBuilder {
+    fn build(&self) -> LinearProgrammingSearch {
+        LinearProgrammingSearch::new(self.game.clone())
+    }
+}
+
+/// Search strategy using ideas from linear programming to order next vertices,
+/// based on distance from the vertex to a region that borders the line between true/false in the formula.
+pub struct LinearProgrammingSearch {
+    /// Priority based on distance, lowest distance highest priority
+    queue: PriorityQueue<Edge<AtlVertex>, i32>,
+    game: IntermediateLcgs,
+    phi_mapper: ConstrainedPhiMapper,
+    /// Maps the hash of a Phi and usize of state, to a result state and distance
```

Comment:
Is this correct?

Reply:
Technically, yes. But it is a copy-paste error too.